lualib/resty/kong/var.lua

```diff
@@ -159,6 +166,16 @@ local function patch_functions()
     variable_index.args = nil
     return orig_set_uri_args(...)
   end
+  
+  local orig_set_header = req.set_header
+
+  req.set_header = function(...)
+    local normalized_header = replace_dashes_lower(...)
```

Comment:
perhaps I need to unpack the data structure(…) and retrieve the first element; this should be clear.

Reply:
right, doesn't the current implementation mean we're passing both `header_name` and `header_value` from `ngx.req.set_header(header_name, header_value)`, to `replace_dashes_lower`?

We can probably patch this with a function that takes exactly 2 arguments, then use the first (header_name) to invalidate the index.

Reply:
I changed to
 
```lua
  req.set_header = function(name, value)
    local normalized_header = replace_dashes_lower(name)
```

 I did not choose the

```lua
  req.set_header = function(...)
    local args = table.unpack(...)
    local normalized_header = replace_dashes_lower(args[1])
```

I feel that this will degrade performance.